src/main/java/seedu/address/logic/commands/ClearCommand.java

```diff
@@ -13,13 +13,19 @@
 
     public static final String COMMAND_WORD = "clear";
     public static final String MESSAGE_SUCCESS = "Address book has been cleared!";
+    public static final String MESSAGE_FAILURE = "No data to clear!";
 
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.setAddressBook(new AddressBook());
-        Summary summary = new Summary(model.getAddressBook());
-        return new CommandResult(MESSAGE_SUCCESS, summary);
+        if (model.getAddressBook().equals(new AddressBook())) {
+            Summary summary = new Summary(model.getAddressBook());
+            return new CommandResult(MESSAGE_FAILURE);
```

Comment:
I was wondering whether it would be better to use `CommandException` to signal that there was no data to clear and hence no action done. _Example of usage: `DeleteCommandIndex#execute()`_

Reply:
Sounds like a better approach. Will do the relevant changes.